File: app/src/main/java/com/donald/musictheoryapp/Music/MusicXML/Score.java
package com.donald.musictheoryapp.Music.MusicXML;

import androidx.annotation.NonNull;

import com.donald.musictheoryapp.Music.ScoreView.MusicXMLParser;

import org.xmlpull.v1.XmlPullParserException;

import java.io.ByteArrayInputStream;
import java.io.IOException;
import java.io.InputStream;

public class Score
{
	public final Part[] parts;

	public Score(Part[] parts) { this.parts = parts; }

	public static Score fromXML(InputStream in) throws IOException, XmlPullParserException
	{
		return new Score(MusicXMLParser.parse(in));
	}

	public static Score fromXML(String xml) throws IOException, XmlPullParserException
	{
		return new Score(MusicXMLParser.parse(new ByteArrayInputStream(xml.getBytes())));
	}

	public Part[] parts() { return parts; }

	@NonNull
	public Score clone()
	{
		Part[] otherParts = new Part[parts.length];
		for(int i = 0; i < otherParts.length; i++)
		{
			otherParts[i] = parts[i].clone();
		}
		return new Score(otherParts);
	}
}

File: app/src/main/java/com/donald/musictheoryapp/music/musicxml/Score.kt
package com.donald.musictheoryapp.music.musicxml

import kotlin.Throws
import org.xmlpull.v1.XmlPullParserException
import com.donald.musictheoryapp.customview.scoreview.MusicXmlParser
import org.dom4j.Document
import org.dom4j.DocumentHelper
import java.io.ByteArrayInputStream
import java.io.IOException
import java.io.InputStream

class Score(val parts: Array<Part>) {

    fun parts(): Array<Part> {
        return parts
    }

    fun clone(): Score {
        val otherParts = Array(parts.size) { index ->
            parts[index].clone()
        }
        return Score(otherParts)
    }

    fun toDocument(): Document {
        val score = DocumentHelper.createDocument()
        score.addElement("score-partwise").apply {
            addAttribute("version", "4.0")
            for (part in parts) {
                part.addToXml(this)
            }
        }
        return score
    }

    companion object {
        @Throws(IOException::class, XmlPullParserException::class)
        fun fromXml(inputStream: InputStream): Score {
            return Score(MusicXmlParser.parse(inputStream))
        }

        @JvmStatic
		@Throws(IOException::class, XmlPullParserException::class)
        fun fromXml(xml: String): Score {
            return Score(MusicXmlParser.parse(ByteArrayInputStream(xml.toByteArray())))
        }
    }

}